packages/datadog-instrumentations/src/helpers/instrument.js

```diff
@@ -28,12 +30,22 @@ exports.addHook = function addHook ({ name, versions, file }, hook) {
       return moduleExports
     }
 
-    try {
-      return hook(moduleExports)
-    } catch (e) {
-      log.error(e)
-      return moduleExports
+    if (moduleName === fullFilename) {
+      const version = getVersion(moduleBaseDir)
```

Comment:
`getVersion` is reading the version from the package.json. We are calling to `getVersion` in line 29, that means that we are reading and parsing `package.json` file twice.
I propose something like that:
```suggestion
   let version
    if (moduleName !== fullFilename || !matchVersion((version = getVersion(moduleBaseDir), version), versions)) {
      return moduleExports
    }

    if (moduleName === fullFilename) {
      if (matchVersion(version, versions)) {
```

Reply:
I wanted to move the code but instead I ended up just copying it, fixing now.